### PR TITLE
Understanding layer 2 blockchain

### LINKED ISSUE #6271: [Blockchain] Understanding layer 2 blockchain
#### Proposed title of article
Understanding layer 2 blockchain

#### Proposed article introduction
Since the release of Bitcoin, blockchains have continued to grow and gain popularity rapidly over the years. People have used the technology in the wide areas apart from just implementing the payments systems. The reason is that blockchain technology is decentralized and highly secure.
It is evident that no single blockchain technology can provide decentralization, security and scalability in achieving the optimal functionality, most tend to sacrifice dome of the elements. A good example is between the Bitcoin and Etherum which competes in terms of scalability, security and decentralization. In order to gain more popularity solutions like layer 2 protocols have appeared in the recent years.
In the proposed article we will cover in more details what is layer 2 blockchain, how it works and why is it needed in the blockchain network.

#### Key takeaways 
At the end of the article the learner will ne able to understand:
- the concept of layet 2 blockchain
- different layer 2 blockchain solutions currently used in the blockchain network
- the reasons why layer 2 blockchains are implemented in the blockchain ecosystem

#### Article quality
The proposed article provides in-depth knowledge and exploration of the layer 2 blockchain and give examples of where it is currently used, unlike the similar articles online which tend to concentrate more on comparisons between layer 1 and layer 2 blockchains and hence missing out on vital details.

#### References
N/A

### PR DESCRIPTION
**Pull request by martha-ngugi**
## Our **Peer Reviewer and Content Moderator teams do NOT provide any revisions services.** All revisions and edits should be completed on your own forked repo (as to not take up room in the queue) in order for our team to review them all in a timely manner. 

**ANY ARTICLE SUBMITTED WITH GLARING ERRORS WILL BE IMMEDIATELY CLOSED.**

>As a rule of thumb - please be sure to only submit articles (pull requests) that are fully polished and ready to be published. Be sure to go through our resources documents for extra and 3rd party (vetted) resources to help improve overall technical writing.

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is **600x400 pixels**, under **300KB in size**, and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

`This closes #6271

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

**Comment by WanjaMIKE** [comment]
Plagiarism check is done

**Comment by martha-ngugi** [comment]
Hi @collins-a, Thank you for taking the time to review my article. Is there any progress on this PR?

File: content/articles/understanding-layer-2-blockchain/index.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+#### Rollups
+These are layer-2 protocols that perform computations off the primary chain. In this blockchain, transaction details transfer occurs after a given time interval. This assists in maintaining records.
+
+Also, rollups handle transactions without interfering with the primary layer. This leads to higher throughput at minimal transaction costs. Since transaction details of the rollups are stored on the primary layer, the rollup's security can be guaranteed.
+
+#### Optimistic rollups
+Optimistic rollups regard that every transaction performed on the blockchain is valid. However, in most cases, optimistic rollups take a considerate amount of time to confirm the transactions. This waiting period gives rollups ample time to resolve a dispute if a challenge arises.
+
+#### Zero-knowledge rollups
+They perform computations off the chain before submitting the primary chain's validity proof. They at times use the [smart contract](https://www.ibm.com/topics/smart-contracts) to hold the funds on the base layer. Funds are then released once the proof of validity has been submitted. The primary chain has also validated and confirmed the transaction.
```
Comment (collins-a) [inline content/articles/understanding-layer-2-blockchain/index.md:71]: Seems like either the last two lines are one line interrupted by a full stop or the last line feels a bit out of place. Rephrase it to convey the intended message.

**Review by collins-a — CHANGES_REQUESTED** [review]
@martha-ngugi good article. Just a couple of considerations to make and we'll be good to go. Ping me when done.

**Review by collins-a — APPROVED** [review]

**Comment by LinusMuema** [comment]
Hi @martha-ngugi 

Great job on the article. Please check my commit and the review changes for future articles.

cc: @collins-a 

🚀

**Comment by martha-ngugi** [comment]
Hello @LinusMuema, The final changes made to the article have been noted. Thank you for reviewing the article.
cc @collins-a
